api.fbs

```diff
@@ -0,0 +1,144 @@
+namespace proto;
+
+// CreateStreamRequest is sent to create a new stream.
+table CreateStreamRequest {
+  subject           : string (id: 0); // Stream NATS subject
+  name              : string (id: 1); // Stream name (unique per subject)
```

Comment:
I'm not very familiar with flatbuffers, but why not mark more fields as `required`?

Reply:
Required is kind of a very strong guarantee that the field will never change, and the idea of using it very sparingly is one of the lessons learned from protobuf (they removed required fields entirely in proto3, see [1] and the quote below).  Once something is required, it can never be modified without changing _all_ clients at the same time, which defeats the purpose of backwards-compatibility of schema changes.  In fact, I'd say maybe we should go the other way and mark _nothing_ as required here.  I have it on a few key fields but maybe for upgradeability i should remove those.  

> We dropped required fields in proto3 because required fields are generally considered harmful and violating protobuf's compatibility semantics. The whole idea of using protobuf is that it allows you to add/remove fields from your protocol definition while still being fully forward/backward compatible with newer/older binaries. Required fields break this though. You can never safely add a required field to a .proto definition, nor can you safely remove an existing required field because both of these actions break wire compatibility. For example, if you add a required field to a .proto definition, binaries built with the new definition won't be able to parse data serialized using the old definition because the required field is not present in old data. In a complex system where .proto definitions are shared widely across many different components of the system, adding/removing required fields could easily bring down multiple parts of the system. We have seen production issues caused by this multiple times and it's pretty much banned everywhere inside Google for anyone to add/remove required fields. For this reason we completely removed required fields in proto3.

[1] https://github.com/protocolbuffers/protobuf/issues/2497

Reply:
Got it. I'm ok with removing `required` altogether.